### PR TITLE
Use `Option<EvoluError>`, `Option<Owner>`  instead of `null`able values

### PR DESCRIPTION
**Pull request by sectore**
**TL;DR** Needed for `@evolu/solid` (and maybe helpful for other implementations). 

Going this way any implementation for Evolu can decide to express `Option` as `null` or `undefined` or anything else. 

For example: `useOwner` for `React` returns `Owner | null`. No change here. But for `Solid` (see #376) `null` are not valid initial values for Solid stores (see [`createstore`](https://www.solidjs.com/docs/latest/api#createstore)). 

With this PR [useSyncEvoluStore](https://github.com/evoluhq/evolu/blob/f482d43aaafb8190ee67a9a9173a003b101b5a14/packages/evolu-solid/src/useSyncEvoluStore.ts) and [useSyncNullableEvoluStore](https://github.com/evoluhq/evolu/blob/f482d43aaafb8190ee67a9a9173a003b101b5a14/packages/evolu-solid/src/useSyncNullableEvoluStore.ts) can be merged easily into a single `useSyncEvoluStore`.

### CODE REVIEW

File: packages/evolu-common/src/Evolu.ts [hunk 1]
```diff
@@ -731,16 +731,16 @@ const EvoluCommon = Layer.effect(
...
     const appState = yield* _(AppState);
 
     dbWorker.onMessage = (output): void =>
       Match.value(output).pipe(
         Match.tagsExhaustive({
-          onError: ({ error }) => errorStore.setState(error),
+          onError: ({ error }) => errorStore.setState(O.some(error)),
           onQuery,
-          onOwner: ({ owner }) => ownerStore.setState(owner),
+          onOwner: ({ owner }) => ownerStore.setState(O.some(owner)),
```
Comment (sectore) [inline packages/evolu-common/src/Evolu.ts:743]: @steida In case `onOwner` callback expects as parameter  `owner: Owner|null` , then change this line to:

```ts
onOwner: ({ owner }) => ownerStore.setState(O.fromNullable(owner)),
```

**Comment by steida** [comment]
Evolu should have a separate Effect API. I don't want to scare people with Effect; there is no reason to change the current API. Solid can and should wrap Evolu API.

**Comment by sectore** [comment]
> Evolu should have a separate Effect API. I don't want to scare people with Effect; there is no reason to change the current API. Solid can and should wrap Evolu API.

It's an internal change for Effect and will be wrapped by `@evolu/solid` or `@evolu/react` or whatever. Users won't touch it. If `evolu` uses `Effect` internally there is no reason to  stick with `null` or `undefined` if there is a better way with `Option`. Don't be shy to use the full power of `Effect` :wink:

**Comment by steida** [comment]
Let's discuss that on Discord
